packages/embark-deployment/src/index.js

```diff
@@ -70,20 +70,32 @@ class DeployManager {
                 if (typeof result === 'function') {
                   callback = result;
                 }
-                contract._gasLimit = self.gasLimit;
-                contract._skipGasEstimations = self.skipGasEstimations;
-                self.events.request('deploy:contract', contract, (err) => {
-                  if (err) {
-                    contract.error = err.message || err;
-                    if (contract.error === constants.blockchain.gasAllowanceError) {
-                      self.logger.error(`[${contract.className}]: ${constants.blockchain.gasAllowanceErrorMessage}`);
+                if (contract.addressHandler && typeof contract.addressHandler === 'function') {
+                  self.plugins.runActionsForEvent('contract:address:handler', {contract}, (err, address) => {
+                    if (err) {
+                      errors.push(err);
                     } else {
-                      self.logger.error(`[${contract.className}]: ${err.message || err}`);
+                      contract.address = address;
+                      self.logger.info(contract.className.bold.cyan + __(' already deployed at ').green + contract.address.bold.cyan);
```

Comment:
We log the same message in the `contracts-deployer` as well, but only if we actually go through `contract:deploy` event. Unless we move some more things around, this seems the most pragmatic way forward to me outputting this message for the same experience (for the enduser)

Reply:
Pro tip with i18n: it's better to use `%s` and `{{}}` so that it's easier to translate.
Eg: 
```
__('{{contractName}} already deployed at {{address}}', {contractName: contract.className.bold.cyan, address: contract.address.bold.cyan}).green
```
I just realized you just copied the message, but eh, it's still better